packages/firestore/src/lite-api/query.ts

```diff
@@ -154,24 +155,22 @@ export function query<T>(
 
 export function query<T>(
   query: Query<T>,
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  filterOrQueryConstraints: any,
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  ...nonFilters: any
+  filterOrQueryConstraint:
+    | QueryCompositeFilterConstraint
+    | QueryConstraint
+    | undefined,
```

Comment:
why do we need `| undefined` here?

Reply:
`filterOrQueryConstraint` can be undefined in this case `query(collection)`. The code would also work if the parameter is declared optional `filterOrQueryConstraint?`, but it also works if the parameter is not declared optional or `| undefined`. Out of the three, I wanted to be explicit, but could also go with making it optional...

Reply:
👍